yb-voyager/cmd/config.go

```diff
@@ -0,0 +1,467 @@
+}
+
+var allowedArchiveChangesConfigKeys = map[string]bool{
+	"delete-changes-without-archiving": true,
+	"fs-utilization-threshold":         true,
+	"move-to":                          true,
+}
+
+var allowedEndMigrationConfigKeys = map[string]bool{
+	"backup-schema-files":    true,
+	"backup-data-files":      true,
+	"save-migration-reports": true,
+	"backup-log-files":       true,
+	"backup-dir":             true,
+}
+
+// Define allowed nested sections
+var allowedConfigSections = map[string]map[string]bool{
+	"source":                        allowedSourceConfigKeys,
+	"target":                        allowedTargetConfigKeys,
+	"assess-migration":              allowedAssessMigrationConfigKeys,
+	"analyze-schema":                allowedAnalyzeSchemaConfigKeys,
+	"export-schema":                 allowedExportSchemaConfigKeys,
+	"export-data":                   allowedExportDataConfigKeys,
+	"export-data-from-source":       allowedExportDataConfigKeys,
+	"export-data-from-target":       allowedExportDataFromTargetConfigKeys,
+	"import-schema":                 allowedImportSchemaConfigKeys,
+	"import-data":                   allowedImportDataConfigKeys,
+	"import-data-to-target":         allowedImportDataConfigKeys,
+	"import-data-to-source":         allowedImportDataToSourceConfigKeys,
+	"import-data-to-source-replica": allowedImportDataToSourceReplicaConfigKeys,
+	"import-data-file":              allowedImportDataFileConfigKeys,
+	"initiate-cutover-to-target":    allowedInitCutoverToTargetConfigKeys,
+	"archive-changes":               allowedArchiveChangesConfigKeys,
+	"end-migration":                 allowedEndMigrationConfigKeys,
+}
+
+// Define mutually exclusive section groups
+var aliasCommandsPrefixes = [][]string{
+	{"export-data", "export-data-from-source"},
+	{"import-data", "import-data-to-target"},
+}
+
+/*
+initConfig initializes the configuration for the given Cobra command.
+
+	It performs the following steps:
+	 1. Creates a new Viper instance to isolate config handling for the command.
+	 2. Loads the config file if explicitly provided via --config, or defaults to ~/.yb-voyager.yaml.
+	 3. Reads and validates the configuration file for allowed global keys, sections, and section keys.
+	 4. Binds Viper config values to Cobra flags, giving priority to command-line flags over config values.
+	 5. Returns an error if config file reading, validation, or flag binding fails.
+
+	This setup ensures CLI > ENV > Config precedence and allows environment variables
+	prefixed with YB_VOYAGER_ to override config values.
+*/
+func initConfig(cmd *cobra.Command) error {
+	v := viper.New()
+
+	if cfgFile != "" {
+		// Use config file from the flag.
+		v.SetConfigFile(cfgFile)
+	} else {
+		// Find home directory.
+		home, err := os.UserHomeDir()
+		cobra.CheckErr(err)
+
+		// Search config in home directory with name ".yb-voyager" (without extension).
+		v.AddConfigPath(home)
+		v.SetConfigType("yaml")
+		v.SetConfigName(".yb-voyager")
```

Comment:
why the "." at the beginning? It is usually to make the file hidden by default. I don't think we want that. 

How about these options for the default: 
- /HOME/yb-voyager-config.yaml
- /HOME/yb-voyager/config.yaml